Math-Trainer/Math-Trainer/TrainViewController.swift

```diff
@@ -9,24 +9,114 @@ import UIKit
+    private var secondNumber = 0
+    var sign: String = ""
+    var count: Int = 0 {
+        didSet {
+            print("Count: \(count)")
+        }
+    }
+    
+    private var answer: Int {
+        switch type {
+        case .add:
+            return firstNumber + secondNumber
+        case .subtract:
+            return firstNumber - secondNumber
+        case .multiply:
+            return firstNumber * secondNumber
+        case .devide:
+            return firstNumber / secondNumber
         }
     }
 
     // MARK: - Life cycle
     override func viewDidLoad() {
-        super.viewDidLoad()
-        configureButtons(buttonsAnswerCollection + [buttonBack])
+        configureQuestion()
+        configureButtons()
+    }
+    
+    // MARK: - IBActions
+    @IBAction func leftAction(_ sender: UIButton) {
+        check(answer: sender.titleLabel?.text ?? "", for: sender)
     }
 
+    @IBAction func rightAction(_ sender: UIButton) {
+        check(answer: sender.titleLabel?.text ?? "", for: sender)
+    }
+    
     // MARK: - Methods
-    private func configureButtons(_ buttons: [UIButton]) {
-        buttons.forEach { $0.addShadow() }
+    private func configureButtons() {
+        [leftButton, rightButton, backButton].forEach { $0.addShadow() }
+
+        let isRightButton = Bool.random()
+        var randomAnswer: Int
+        
+        repeat {
+            randomAnswer = Int.random(in: (answer - 1)...(answer + 1))
+        } while randomAnswer == answer
+        
+        rightButton.setTitle(isRightButton ? String(answer) : String(randomAnswer), for: .normal)
+        leftButton.setTitle(isRightButton ? String(randomAnswer) : String(answer), for: .normal)
+    }
+    
+    private func configureQuestion() {
+        (firstNumber, secondNumber) = getTwoRandomNumbers()
+
+        if type == .devide {
+            repeat {
+                (firstNumber, secondNumber) = getTwoRandomNumbers()
+            } while isNotCorrectDivisorOrDividend(divisor: firstNumber, dividend: secondNumber)
+        }
+        
+        let question: String = "\(firstNumber) \(sign) \(secondNumber) = "
+        questionLabel.text = question
+    }
+    
+    private func getTwoRandomNumbers() -> (Int, Int) {
+        return (Int.random(in: 1...99), Int.random(in: 1...99))
+    }
+    
+    private func isNotCorrectDivisorOrDividend(divisor: Int, dividend: Int) -> Bool {
+        return (divisor % dividend != 0) || (divisor == dividend) || (divisor == 1 || dividend == 1)
+    }
+    
+    private func check(answer: String, for button: UIButton) {
+        let isRightAnswer = Int(answer) == self.answer
+        
+        button.backgroundColor = isRightAnswer ? .green : .red
+        
+        if isRightAnswer {
+            let isSecondAttempt: Bool = rightButton.backgroundColor == .red || leftButton.backgroundColor == .red
+            count += isSecondAttempt ? 0 : 1
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
```

Comment:
I suggest to disable buttons before this` DispatchQueue.main.asyncAfter` user should not be able to use buttons while this timer runs.
```
if isRightAnswer {
            let isSecondAttempt: Bool = rightButton.backgroundColor == .red || leftButton.backgroundColor == .red
            count += isSecondAttempt ? 0 : 1
            rightButton.isEnabled = false
            leftButton.isEnabled = false
            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
                rightButton.isEnabled = true
                leftButton.isEnabled = true
                self?.configureQuestion()
                self?.configureButtons()
            }
```

Reply:
Yep, thanks, done.